### PR TITLE
Feature: GPT Key Request on the landing page

### LINKED ISSUE #87: Implement GPT Key Request
**Description:**
This tasks inlcudes the implementation of a UI for pasting in the gpt key

# Acceptance Criteria
- [x] in the landing page should be in the right corner anicon (key) or sth. where you can click on it 
- [x] after a pop up appears where you can change your gpt key
- [x] when clicking on app the access should be blocked by the pop up to force the user to enter a key

### PR DESCRIPTION
**Pull request by thangixd**
Hi,
I have added a feature that checks if there is an Open API key in the environment variables. If not, a popup appears forcing the user to enter an API key in order to use TracEX. 
image

If a user submits an empty string, an error message in red will appear, indicating that input is required.
image

I've designed the code to be easily extendable for future validations, such as verifying whether a key is a valid open API key, checking if it still has tokens available, or ensuring it meets specific length requirements.

This PR closes #87.

### CODE REVIEW

**Comment by PitButtchereit** [comment]
Why are we working on low priority tasks and why has our PO not approved this but it is already in the PR stage?

**Comment by tkv29** [comment]
yep, if someone wants to do more the person is invited to do so
We have enough tasks :)

**Review by nils-schmitt — APPROVED** [review]
Looks good and works fine!

File: tracex_project/tracex/templates/landing_page.html [hunk 1]
```diff
@@ -3,35 +3,70 @@
...
+                    </div>
+                    <div class="modal-footer">
+                        <button type="submit" class="btn btn-secondary">Save API Key</button>
+                    </div>
+                </form>
+            </div>
+        </div>
+    </div>
+</div>
+<script>
```
Comment (tkv29) [inline tracex_project/tracex/templates/landing_page.html:65]: duplicated with the code in your js

File: tracex_project/tracex/static/tracex/css/style.css [hunk 2]
```diff
@@ -339,3 +339,30 @@ text-align: center;
     text-decoration: none;
     cursor: pointer;
 }
+
+.modal-backdrop {
+    z-index: 1040; /* or lower than modal if modal needs to be on top */
+}
+
+.modal {
```
Comment (tkv29) [inline tracex_project/tracex/static/tracex/css/style.css:347]: u can merge this with the .modal that you wrote before

File: tracex_project/tracex/static/tracex/js/forms_open_ai_key.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+(function () {
```
Comment (tkv29) [inline tracex_project/tracex/static/tracex/js/forms_open_ai_key.js:1]: could you please explaing what you doing here in snippet

File: tracex_project/tracex/views.py [hunk 4]
```diff
@@ -1,15 +1,40 @@
 """This file contains the views for the landing page of the tracex app."""
+import os
 from django.views.generic import TemplateView
+from django.shortcuts import render, redirect
+from .forms import ApiKeyForm
```
Comment (tkv29) [inline tracex_project/tracex/views.py:5]: please use absolute paths

**Review by tkv29 — COMMENTED** [review]
added some comments, gj!

File: tracex_project/tracex/static/tracex/js/forms_open_ai_key.js (hunk 3 above)
Comment (thangixd) [reply tracex_project/tracex/static/tracex/js/forms_open_ai_key.js:1]: ensures that forms with the class needs-validation are checked for validity before submission and visually marked as validated

 If the form is not valid (form.checkValidity() === false), it prevents the form from being submitted (event.preventDefault()) and stops further propagation of the event (event.stopPropagation()).

File: tracex_project/tracex/views.py (hunk 4 above)
Comment (thangixd) [reply tracex_project/tracex/views.py:5]: I dont know why, but I cant figure out why it isnt working with:  

from tracex_project.tracex.forms import ApiKeyForm

I looked up how we imported such things before however we also used . import class and never the absolute path

Comment (thangixd) [reply tracex_project/tracex/views.py:5]: image

not working

Comment (thangixd) [reply tracex_project/tracex/views.py:5]: image

not working

Comment (thangixd) [reply tracex_project/tracex/views.py:5]: nice! than I will just commit it like that

**Comment by tkv29** [comment]
just checked it with docker and it works!
Would be really cool if there is an key icon at the top right of the landing page, so the user can edit the key
Else the user has to restart the container / server

**Review by tkv29 — APPROVED** [review]

**Pull request closed without merging**